Update I18N to use unique id

OlivierJaquemet opened the pull request:
- Update all JS files to use a new unique identifier as the key of
  translated message when invoking I18N methods (instead of english).
  See discussion in issue #1352 for full rational.
  Basically the idea is to simplify translation process.
- In "me-i18n.js", specify english as the default language to use
  for key
  that could not be found in the current context language
- Update all existing localization files to use new UID
- Removed unused message `"Play/Pause"` in I18N files
- mep-feature-ads : Update I18N message to use one I18N property
  for sentence "Skip in %1 seconds" instead of using string
  concatenation
- mep-feature-ads : Add missing message in I18N files

   `'mejs.ad-skip': 'Skip ad'`
   `'mejs.ad-skip-info': 'Skip in %1 seconds'`

Tests :
- Include english translation file in the test samples
  (as it now required)

Build :
- Change default build to only include the english
  translation (this is an abitrary decision, and could be
  modified easiliy)

Migration :

To update any existing JS file to use new keys,
the following sed script can be applied :

`find . -name "*.js" -exec sed -b -i -f replacement.sed "{}" \;`

```
s;["']Play["'];'mejs.play';g
s;["']Pause["'];'mejs.pause';g
s;["']None["'];'mejs.none';g
s;["']Close["'];'mejs.close';g
s;["']Download File["'];'mejs.download-file';g
s;["']Download Video["'];'mejs.download-video';g
s;["']Captions/Subtitles["'];'mejs.captions-subtitles';g
s;["']Fullscreen["'];'mejs.fullscreen';g
s;["']Go Fullscreen["'];'mejs.fullscreen-on';g
s;["']Turn off Fullscreen["'];'mejs.fullscreen-off';g
s;["']Mute["'];'mejs.mute';g
s;["']Mute Toggle["'];'mejs.mute-toggle';g
s;["']Unmute["'];'mejs.unmute';g
s;["']Jump forward %1 seconds["'];'mejs.time-jump-forward';g
s;["']Skip back %1 seconds["'];'mejs.time-skip-back';g
s;["']Time Slider["'];'mejs.time-slider';g
s;["']Use Left/Right Arrow keys to advance one second, Up/Down arrows to advance ten seconds.["'];'mejs.time-help-text';g
s;["']Use Up/Down Arrow keys to increase or decrease volume.["'];'mejs.volume-help-text';g
s;["']Volume Slider["'];'mejs.volume-slider';g
s;["']Audio Player["'];'mejs.audio-player';g
s;["']Video Player["'];'mejs.video-player';g
s;["']Skip ad["'];'mejs.ad-skip';g
s;["']Skip in %1 seconds["'];'mejs.ad-skip-info';g
```

PS : being unfamiliar with git, this PR is a continuation of initial PR discussed here : 
https://github.com/johndyer/mediaelement/pull/1413

ron666 commented on src/js/me-i18n.js:
```diff
@@ -40,6 +37,7 @@
     "use strict";
 
     var i18n = {
+        "default_language": 'en',
```
I'll use only `default` since it's understood that is part of the `i18n` object for more simplicity

OlivierJaquemet replied:
fixed in 	a757d06

ron666 commented on src/js/mep-feature-googleanalytics.js:
```diff
@@ -9,8 +9,8 @@
 	googleAnalyticsCategory: 'Videos',
-	googleAnalyticsEventPlay: 'Play',
-	googleAnalyticsEventPause: 'Pause',
+	googleAnalyticsEventPlay: 'mejs.play',
```
I don't know if this feature really requires translation, given that not all of the elements listed are translated. Thoughts?

ron666 commented on src/js/mep-feature-universalgoogleanalytics.js:
```diff
@@ -8,8 +8,8 @@
 	googleAnalyticsCategory: 'Videos',
-	googleAnalyticsEventPlay: 'Play',
-	googleAnalyticsEventPause: 'Pause',
+	googleAnalyticsEventPlay: 'mejs.play',
```
I don't know if this feature really requires translation, given that not all of the elements listed are translated. Thoughts?

OlivierJaquemet replied:
Good catch! this would clearly be a regression. 

Please help with Git/GitHub/PR best practices... I'm a noob here : Should I add a new commit with a fix ? or somehow rewrite my commit in order to exclude this file, and which way would you recommend ?

ron666 replied:
Just add a new commit; rewriting a commit is difficult and potentially dangerous for the repo integrity

ron666 commented:
@OlivierJaquemet I left some comments on this PR. It looks good; only consider those comments and also the skip back and forward translations are available in English and Spanish. Is there a way we can integrate them in the other language files to have all the files completed as the English one?

OlivierJaquemet commented:
Regarding missing translations : 
- I just fixed the french one (in which I had also introduced a regression in my initial commit). 
- I'll send all the missing bits to our translators, and will be able to provide them propably in october. 

In the meantime, here are some statistics regarding current transalation progress in files bundled in the repository (those stats are valid after my last commit). As you can see, only the english and french are fully translated.
![image](https://cloud.githubusercontent.com/assets/618861/18782314/347bf84e-8186-11e6-8224-0061235ccd69.png)

ron666 commented:
Thanks for the stats. I'll wait for the rest of the translations to merge this PR so we have a complete and pretty solid change with this. Good job!

OlivierJaquemet commented:
@ron666 New translation added in last commit, which bring us to the following stats : 
![image](https://cloud.githubusercontent.com/assets/618861/18786394/ed5450de-819d-11e6-8299-480be2062632.png)

Given the low number of missing translation, and also because none are being bundled in the build ... Would you consider merging the PR now ? This would prevent potential PR conflict hard to resolve if I get the remaining translation long after other important commit in the master ...

ron666 commented:
Excellent. So this approach won't break anything on Wordpress correct?

OlivierJaquemet commented:
It WILL break things. But @ocean90 is aware of it gave us his agreement in the following comment : https://github.com/johndyer/mediaelement/pull/1413#issuecomment-245542486

The next modification regarding I18N will be to properly implement plural/gender and such, to fix #1792 . However this modification will also require important decision on which format to use (ICU msg format ? other ) and how to implement it (rely on external project?). But I think this discussion does not belong in this PR.

ocean90 commented on src/js/me-i18n.js:
```diff
@@ -107,13 +105,22 @@
         // Fetch the localized version of the string.
-        if (i18n.locale.strings && i18n.locale.strings[options.context] && i18n.locale.strings[options.context][str]) {
-            str = i18n.locale.strings[options.context][str];
+        if (i18n.locale.strings && i18n.locale.strings[options.context]) {
```
Why don't you check if `i18n.locale.strings[options.context][uid]` exists like before with `str`?

OlivierJaquemet replied:
Because it would be a redudant check with the following line which checks str
https://github.com/jalios/mediaelement/blob/2c5f19d56bc494f895ab1af97b9317b01d7d9536/src/js/me-i18n.js#L117

ron666 commented:
OK if no more comments from @ocean90 I'll merge this tonight.

OlivierJaquemet commented:
Thank you @ron666 for your review and help with Git. 
We (at Jalios) are glad to participate in this project which allows us to provide nice feature for our users.
